app/views/layouts/application.html.erb

```diff
@@ -0,0 +1,79 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title><%= content_for?(:title) ? yield(:title) : "Media Ranker" %></title>
+    <link rel="shortcut icon" type="image/x-icon" href="https://media-ranker-2-0.herokuapp.com/assets/favicon-910822ff6cd5f6d37119171c4b716f1d4e5fe0e8ad5e408627659ebf8d7f16b1.ico", alt: "site logo, an owl">
+    <link href="https://fonts.googleapis.com/css?family=Gudea" rel="stylesheet">
+    <%= stylesheet_link_tag    "application" %>
+    <%= javascript_include_tag "application", 'data-turbolinks-track' => true %>
+    <%= csrf_meta_tags %>
+  </head>
+  <body>
+    <header>
+      <div >
+        <h2 class="hide-for-small-only"><%= link_to "Media Ranker", root_path %><small>Ranking the Best of Everything</small></h2>
+      </div>
+      <div class="title-bar" data-responsive-toggle="main-nav" data-hide-for="medium">
+        <h2>
+          <span>
+          <button class="menu-icon" type="button" data-toggle="main-nav">
+          </button>
+        </span>
+        <div class="title-bar-title">
+        <%= link_to "Media Ranker", root_path %>
+        </div>
+        </h2>
+      </div>
+      <div class="top-bar" id="main-nav">
+        <div class="top-bar-left">
+          <ul class="dropdown vertical medium-horizontal menu" data-responsive-menu="drilldown medium-dropdown">
+            <li><%= link_to "View top media", root_path, class: "button" %></li>
+            <li><%= link_to "View all media", works_path, class: "button" %>
+            <li><%= link_to "Add a new work", new_work_path, class: "button" %></li>
+            <li><%= link_to "View all users", users_path, class: "button" %></li>
+          </ul>
+        </div>
+        <div class="top-bar-right">
+          <ul class="menu">
+            <% if session[:user_id] != nil %>
+              <li><%= link_to "Logged in as #{User.get_username(session[:user_id])}" , user_path(session[:user_id]), method: :get, class: "button"%></li>
```

Comment:
You reference `session[:user_id]` a lot in this section of code:
- you use a conditional on it
- you get the username here through `{User.get_username(session[:user_id])}`
- you use it for the user_path param

You already declare `before_action :find_user` in each controller, so each view will have a value on `@user`... maybe you could use that value?